File: 35.搜索插入位置.py
#
# @lc app=leetcode.cn id=35 lang=python
#
# [35] 搜索插入位置
#

# @lc code=start
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        if target in nums:
            return nums.index(target)
        if target > nums[-1]:
            return len(nums)
        if target < nums[0]:
            return 0
        for i in range(len(nums) - 1):
            if nums[i] < target and nums[i+1] > target:
                return i+1
# @lc code=end

File: 35.搜索插入位置.cpp
/*
 * @lc app=leetcode.cn id=35 lang=cpp
 *
 * [35] 搜索插入位置
 */

// @lc code=start
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int right = nums.size();
        if(target < nums[0])
            return 0;
        else if(target > nums[right-1])
            return right;
        int left = 0; 
        while(left < right){
            int mid = (left + right) /2;
            if(nums[mid] == target or (nums[mid] > target and nums[mid-1] < target))
                return mid;
            else if(nums[mid] > target){
                right = mid-1;
            }
            else{
                left = mid+1;
            }
        }
        return right;

    }
};
// @lc code=end

